rir/src/ir/RuntimeFeedback.h

```diff
@@ -33,9 +35,43 @@ struct CallFeedback {
     }
 };
 
+struct RecordedType {
+    uint8_t sexptype : 5;
+    uint8_t scalar : 1;
+    uint8_t object : 1;
+    uint8_t attribs : 1;
+    RecordedType() {}
+    RecordedType(SEXP s);
+    bool operator==(const RecordedType& other) {
+        return memcmp(this, &other, sizeof(RecordedType)) == 0;
+    }
+};
 static_assert(sizeof(CallFeedback) == 7 * sizeof(uint32_t),
               "Size needs to fit inside a record_ bc immediate args");
 
+struct TypeFeedback {
+    static constexpr unsigned MaxTypes = 3;
+    uint8_t numTypes;
+
+    std::array<RecordedType, MaxTypes> seen;
+
+    TypeFeedback() {}
+
+    void record(SEXP e) {
+        RecordedType type(e);
```

Comment:
OK, so I was wondering how this was working and if there was some weird C++ black magic... but turns out it isn't working -- the constructor isn't defined!